### PR TITLE
Include database

### PR DESCRIPTION
**Pull request by susiecoleman**
First pass at how collecting metrics will work
- Included database with quill and some basic operations
- Did a bit of thinking about how to do json parsing on the side

### CODE REVIEW

**Comment by gtrufitt** [comment]
This lgtm, but maybe someone with better Scala eyes like @paulmr or @jennysivapalan could take a gander? :)

File: app/util/Utils.scala [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+import io.circe.ParsingFailure
+import models.{InvalidJsonError, NoRequestBodyError, ProductionMetricsError, UnexpectedExceptionError}
+
+object Utils {
+
+  def processException(exception: Exception): Either[ProductionMetricsError, Nothing] = {
+    val error = exception match {
+      case e: ParsingFailure => InvalidJsonError(e.message)
+      case _ => UnexpectedExceptionError
```
Comment (jennysivapalan) [inline app/util/Utils.scala]: It looks like you'll be hiding the exception here and instead reporting something unexpected happened. I'm not sure how useful that would be, you'll have the message in the `Exception` so you could log it out.

File: app/controllers/App.scala [hunk 2]
```diff
@@ -1,14 +1,27 @@
...
+    Logger.info(s"I am the ${config.appName}")
     Ok(views.html.index())
   }
+
+  def createdInCopyContent = AuthAction { req =>
+    val x = for {
+      body <- extractRequestBody(req.body.asJson.map(_.toString))
+      json <- stringToJson(body)
+      data <- jsonToInCopyData(json)
+    } yield data
```
Comment (paulmr) [inline app/controllers/App.scala]: This is too clever for me! I have tried to work out what we are doing here but I can't.

Are we just running this for the side-effect of throwing an exception if it is not valid?

Could we make that more explicit by just doing something like:

```scala
for {
  body <- extractRequestBody(req.body.asJson.map(_.toString))
  json <- stringToJson(body)
  _  <- jsonToInCopyData(json)
} yield ()
```

as a side note, I can't figure out why this won't compile if I change it to:

```scala
for {
  body <- extractRequestBody(req.body.asJson.map(_.toString))
  json <- stringToJson(body)
  _  <- jsonToInCopyData(json)
}
```

I would love to know why if you know!

File: app/config/Config.scala [hunk 3]
```diff
@@ -18,24 +17,16 @@ object Config extends AwsInstanceTags {
...
+    .getOrElse(throw new RuntimeException(s"key $key not found in configuration"))
+
+  val elkKinesisStream = getConfigString("elk.kinesis.stream")
+  val elkLoggingEnabled = config.getBoolean("elk.logging.enabled").getOrElse(true)
+
+  val pandaDomain = getConfigString("panda.domain")
+  val pandaAuthCallback = getConfigString("panda.authCallback")
+  val pandaSystem = getConfigString("panda.system")
+
+  val dbConfig = config.underlying.getConfig("ctx")
```
Comment (susiecoleman) [inline app/config/Config.scala]: `underlying` needs to be used here as the dbConfig is passed into the `PostgresJdbcContext` object in the `MetricsDB`. `PostgresJdbcContext` requires something of type `com.typesafe.config.Config`  not a `play.api.Configuration`

File: app/controllers/App.scala (hunk 2 above)
Comment (susiecoleman) [inline app/controllers/App.scala]: Sorry this is a slightly strange bit of code I was experimenting with parsing so the function name is not very helpful. I would expect this to put the data into the database but I didn't ever connect up the dots in this PR as I wanted to get feedback on the quill work. It probably didn't need to go in this PR

File: app/util/Utils.scala (hunk 1 above)
Comment (susiecoleman) [inline app/util/Utils.scala]: Added some logging

**Comment by paulmr** [comment]
Thanks for replying @susiecoleman 
This looks good to me.

**Pull request merged**